src/client/caching/structures/Message.ts

```diff
@@ -192,10 +195,32 @@ export class Message extends Structure {
 		return this.editedTimestamp ? new Date(this.editedTimestamp) : null;
 	}
 
+	/**
+	 * Edits the message.
+	 * @param content The {@link MessageBuilder builder} to send.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#edit-message
+	 */
+	public async edit(content: MessageBuilder): Promise<Message> {
+		const data = await this.client.api.patch(Routes.channelMessage(this.channel.id, this.id), content) as APIMessageData;
+		return this._patch(data) as Message;
+	}
+
+	/**
+	 * Deletes the message.
+	 * @param reason The reason to write in the audit logs.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#delete-message
+	 */
+	public async delete(reason: string): Promise<this> {
```

Comment:
should internally call MessageStore#remove() with this code. And also mark the message as deleted as previously requested to avoid race conditions.